### DISCUSSION TITLE
SAR_look

### DISCUSSION THREAD

**Discussion by Simon202121** [gmtsar/gmtsar#951] [Q&A] [answered]
Hi，I understand that SAR_look command need an inputfile which includes Longitude, latitude, elevation.but how can I get those information by a grid file?

**Comment by Xiaohua-Eric-Xu** [comment, marked as answer]
You could use `gmt grdtrack` to extract elevation information. e.g.,\
```
gmt grdtrack lonlat.dat -Gdem.grd > data.llt
```

**Reply by Simon202121** [reply]
Thank you so much！Your answer is just right. I'm very sorry because I have annother question to ask you.
When I got 'los_ll.grd' and generrated image for analysing .I found many NAN pixel on the image，which makes it look rough like this.
![e076553ca6386d83a58edee425840e2](https://github.com/user-attachments/assets/c8d770da-1eba-48d2-a6a7-19f127b2d241)
Could you tell me how can I make it look smooth?I have considered many factors，such as filter_wavelength, decimation of images and interpolation it.But I don't know which one really works.

**Reply by hillgrey** [reply]
I met the same problem as you posted here. I understand there are areas with NAN values and this is quite normal for low coherent areas, but the holes of every pixel with NAN value seem rather big, and they look like strips.
Also, I don't understand why it looks like this and how to solve it.
I guess this is related to the mismatch between the "filter_wavelength" and the range/azimuth looks, because I have tried several combinations of these settings, and filter_wavelength=200, range_dec=8, azimuth_dec=2, worked well most of the time.
But, what if I wish to use other filter_wavelength and multilook factors? I don't know.

**Reply by Xiaohua-Eric-Xu** [reply]
Part of this comes from a known effect for side looking radars. Another could be due to decorrelation. 
![image](https://github.com/user-attachments/assets/823f50c8-4e3c-48c4-a824-269702e672e8)
For the former, you can reduce projection resolution (or increase pixel size) so these gaps are gone (though it does not mean they are properly measured, only smoothing through). Another would be to use gmt grdfilter -Ni to interpolate through.
